ui/hooks/useNftCollectionsMetadata.ts

```diff
@@ -0,0 +1,64 @@
+import { useEffect, useMemo, useState } from 'react';
+import { NameType } from '@metamask/name-controller';
+import { Collection } from '@metamask/assets-controllers';
+import { useSelector } from 'react-redux';
+import { getCurrentChainId } from '../selectors';
+import { fetchNftCollectionsMetadata } from '../store/actions';
+
+export type UseNftCollectionsMetadataRequest = {
```

Comment:
Are there any cases where we can know the contract address and token id, or will we always just know the contract address?

Reply:
Exactly, we know `tokenId` and even the token standard either `ERC721` or `ERC1155`. And it totally make sense to me go for on-chain data before asking to API to reduce amount of requests. 

While I was experimenting the existing functions like [`getERC721TokenURI`](https://github.com/MetaMask/core/blob/main/packages/assets-controllers/src/AssetsContractController.ts#L391) or [`getERC1155TokenURI`](https://github.com/MetaMask/core/blob/main/packages/assets-controllers/src/AssetsContractController.ts#L455) I encountered this error consistently. Pretty sure that I am sending the correct values.

```
Error: missing revert data in call exception; Transaction reverted without a reason string [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (data="0x", transaction={"to":"0xcF89D277B19Eba3752DFe66bAF23a658AB2bA9Cd","data":"0xc87b56dd0000000000000000000000000000000000000000000000000000000000000003","accessList":null}, error={"code":-32000,"message":"execution reverted"}, code=CALL_EXCEPTION, version=providers/5.7.2)
  at Logger.makeError (index.js:238:1)
  at Logger.throwError (index.js:247:1)
  at checkError (json-rpc-provider.js:108:1)
  at Web3Provider.<anonymous> (json-rpc-provider.js:751:1)
  at step (json-rpc-provider.js:48:1)
  at Object.throw (json-rpc-provider.js:29:1)
  at rejected (json-rpc-provider.js:21:1)
```

These errors are ignored in the `NFTController`: https://github.com/MetaMask/core/blob/main/packages/assets-controllers/src/NftController.ts#L707, probably for some edge cases.

But the same errors are also happening on while adding NFTs. 
https://github.com/MetaMask/metamask-extension/assets/7644512/f14b96a9-5791-4f32-9be3-4a778ecdbbb6

Any advise here? @bergeron @sahar-fehri

Reply:
Exactly, we know tokenId and even the token standard either ERC721 or ERC1155. And it totally make sense to me go for on-chain data before asking to API to reduce amount of requests.

I've reused the `getNftInformationFromTokenURI` from `NFTController`, now this hook primarily use blockchain data, and it will fallback into NFT API calls if it fails. 

This is purely implemented in this commit to make clear separation from other changes here: https://github.com/MetaMask/metamask-extension/pull/25692/commits/96e71b53c16f85c289e7eeb3125dff1c59e41ce4

Let me know if we can do further optimisations.

Reply:
I'm not sure I follow the logic here and I fear we're starting to tangent from the original use case.

The single goal here is to display default petnames for contract addresses that are NFTs and that we can determine are "trusted" using the `isSpam` property from Reservoir.

We can't determine that on chain, so any additional logic to request data via the provider seems unnecessary at this stage?

Reply:
Wrapping up the latest implementation in this PR after discussion;
- Using `getTokenStandardAndDetails` with only given contract address to determine the `standard` of the token
- If it is `ERC721` then it calls the api and fetch collection info
- Depending on the `isSpam` property it decides to use it or not.